### ISSUE TITLE
Segfault on <Image> load after a few hot reloads or manual reloads

### ISSUE THREAD

**Issue by asciifaceman** [facebook/react-native#18261]
I first posted this issue to the Expo team [Here](https://github.com/expo/expo/issues/1487), so I am copy pasta + linking

The tl;dr high level is that if I utilize `<Image>` JSX and load an image then the iOS simulator will Segfault (all detailed below) after a few Hot or Manual reloads. If I remove the `<Image>` then I can sit there and hold CMD+R all day long without issue (it sounds silly, but I sat there and held it and reloaded for about 2 minutes, probably 40-50 reloads just to make sure - with the Image tag it segfaults within 3-5 reloads. Sometimes 10 if I freshly restart everything).

### Environment

```
Environment:
  OS: macOS Sierra 10.12.6
  Node: 6.11.3
  Yarn: 1.1.0
  npm: 3.10.10
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  react: 16.2.0 => 16.2.0
  react-native: https://github.com/expo/react-native/archive/sdk-25.0.0.tar.gz => 0.52.0
```

```
15:16 $ exp diagnostics
[exp] There is a new version of exp available (49.2.2).
You are currently using exp 47.1.1
Run `npm install -g exp` to get the latest version
[exp] Generating diagnostics report...
[exp] Please share this URL with the Expo team: https://exp-xde-diagnostics.s3.amazonaws.com/flanldev-1bcbac19-fbed-41da-8011-777e7e7193f6.tar.gz.
```

Testing on iOS 11.2 on iPhone 7+
Alternative tests performed on Android Nexus 6 physical, and LG V30 physical (no access to physical iPhone at the moment). 

<iOS, Android>

### Expected Behavior

Code example:

```
      <View style={styles.container}>
        <View style={styles.logoContainer}>
          <Image 
            ref="logo" 
            source={require('../assets/photo.png')}
            style={styles.logo}
          />
        </View>
      </View>
```

### Actual Behavior

```
Mar  1 19:42:40 bonfire com.apple.CoreSimulator.SimDevice.56F9EA55-B2EF-429F-81E1-A92107E63452[7346] (UIKitApplication:host.exp.Exponent[0xd9b6][7372][7517]): Service exited due to signal: Segmentation fault: 11 sent by exc handler[0]
Mar  1 19:42:40 bonfire com.apple.CoreSimulator.SimDevice.56F9EA55-B2EF-429F-81E1-A92107E63452[7346] (com.apple.WebKit.Networking.6A03FC1B-DDDE-405C-A357-B8C00108DAF5[7519]): Service exited with abnormal code: 1
Mar  1 19:42:40 bonfire launchd_sim[7346]: BUG in libdispatch client: kevent[mach_recv] monitored resource vanished before the source cancel handler was invoked
```

App exits, no further stack trace or errors are reported. 
Problem disappears if the `<Image>` is removed. This persists across different images and across packager reloads / node_modules being rebuilt.

### Steps to Reproduce

I've been unable to reproduce on a physical LG V30 device. However the ios simulator guaranteed within 10 hot reloads will segfault.

Setting up a simple view and loading an image and then manually or hot reloading reproduces for me. Setting width and height to undefined does not crash, but of course the image is not loading either.
Absolute positioning without width/height also crashes.

I'll update if I missed anything, lots to unpack here.

Addendum:

I've tried to utilize size naming (*@2x.png *@3x.png etc) as well
As a test I also set width & height to `undefined` and the crash went away. But any time the image actually renders, the crash occurs. (any image, I'm using the image colloquially)

**Comment by hramos** [comment]
Feel free to ignore the warning above, as 0.52 is the most recent version you can use with Expo.

**Comment by asciifaceman** [comment]
Being fair I was unable to reproduce so far on ejected 0.53, but that's an
entirely different layout and world
